File: sdk/java/examples/UsageExample.java
package com.hjtpx.captcha.examples;

import com.hjtpx.captcha.client.CaptchaClient;
import com.hjtpx.captcha.model.*;

import java.util.Arrays;
import java.util.List;

public class UsageExample {

    public static void main(String[] args) {
        String baseUrl = "http://localhost:8080";
        String apiKey = "your-api-key";
        String secretKey = "your-secret-key";

        try (CaptchaClient client = new CaptchaClient(baseUrl, apiKey)) {
            sliderCaptchaExample(client);
            clickCaptchaExample(client);
            rotationCaptchaExample(client);
            gestureCaptchaExample(client);
            jigsawCaptchaExample(client);
            voiceCaptchaExample(client);
            connectCaptchaExample(client);
            threeDCaptchaExample(client);
            loginExample(client);
            detectionExample(client);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    private static void sliderCaptchaExample(CaptchaClient client) {
        System.out.println("=== Slider Captcha Example ===");
        try {
            SliderCaptchaResponse captcha = client.getSliderCaptcha(320, 160, 5);
            System.out.println("Session ID: " + captcha.getSessionId());
            System.out.println("Image URL: " + captcha.getImageUrl());
            System.out.println("Puzzle URL: " + captcha.getPuzzleUrl());
            System.out.println("Secret Y: " + captcha.getSecretY());

            TrajectoryPoint point1 = new TrajectoryPoint(0, captcha.getSecretY(), System.currentTimeMillis() - 1000);
            TrajectoryPoint point2 = new TrajectoryPoint(50, captcha.getSecretY() + 5, System.currentTimeMillis() - 800);
            TrajectoryPoint point3 = new TrajectoryPoint(100, captcha.getSecretY() - 3, System.currentTimeMillis() - 500);
            TrajectoryPoint point4 = new TrajectoryPoint(150, captcha.getSecretY() + 2, System.currentTimeMillis() - 200);
            TrajectoryPoint point5 = new TrajectoryPoint(180, captcha.getSecretY(), System.currentTimeMillis());
            List<TrajectoryPoint> trajectory = Arrays.asList(point1, point2, point3, point4, point5);

            VerifyCaptchaResponse verifyResponse = client.verifySliderCaptcha(
                captcha.getSessionId(),
                180,
                captcha.getSecretY(),
                trajectory
            );

            System.out.println("Verification success: " + verifyResponse.isSuccess());
            System.out.println("Message: " + verifyResponse.getMessage());
            if (verifyResponse.getTrajectoryResult() != null) {
                System.out.println("Trajectory score: " + verifyResponse.getTrajectoryResult().getScore());
            }
        } catch (Exception e) {
            System.err.println("Slider captcha error: " + e.getMessage());
        }
    }

    private static void clickCaptchaExample(CaptchaClient client) {
        System.out.println("\n=== Click Captcha Example ===");
        try {
            ClickCaptchaResponse captcha = client.getClickCaptcha("number", true, 3);
            System.out.println("Session ID: " + captcha.getSessionId());
            System.out.println("Image URL: " + captcha.getImageUrl());
            System.out.println("Hint: " + captcha.getHint());
            System.out.println("Hint order: " + captcha.getHintOrder());

            List<List<Integer>> points = Arrays.asList(
                Arrays.asList(50, 50),
                Arrays.asList(150, 100),
                Arrays.asList(250, 80)
            );

            VerifyCaptchaResponse verifyResponse = client.verifyClickCaptcha(
                captcha.getSessionId(),
                points,
                captcha.getHintOrder()
            );

            System.out.println("Verification success: " + verifyResponse.isSuccess());
            System.out.println("Message: " + verifyResponse.getMessage());
        } catch (Exception e) {
            System.err.println("Click captcha error: " + e.getMessage());
        }
    }

    private static void rotationCaptchaExample(CaptchaClient client) {
        System.out.println("\n=== Rotation Captcha Example ===");
        try {
            RotationCaptchaResponse captcha = client.getRotationCaptcha();
            System.out.println("Challenge ID: " + captcha.getChallengeId());
            System.out.println("Image URL: " + captcha.getImageUrl());

            int correctAngle = 180;
            VerifyCaptchaResponse verifyResponse = client.verifyRotationCaptcha(
                captcha.getChallengeId(),
                correctAngle
            );

            System.out.println("Verification success: " + verifyResponse.isSuccess());
            System.out.println("Message: " + verifyResponse.getMessage());
        } catch (Exception e) {
            System.err.println("Rotation captcha error: " + e.getMessage());
        }
    }

    private static void gestureCaptchaExample(CaptchaClient client) {
        System.out.println("\n=== Gesture Captcha Example ===");
        try {
            GestureCaptchaResponse captcha = client.getGestureCaptcha();
            System.out.println("Session ID: " + captcha.getSessionId());
            System.out.println("Pattern: " + captcha.getPattern());
            System.out.println("Grid size: " + captcha.getGridSize());

            List<Integer> pattern = Arrays.asList(1, 2, 3, 5, 7, 8, 9);
            VerifyCaptchaResponse verifyResponse = client.verifyGestureCaptcha(
                captcha.getSessionId(),
                pattern
            );

            System.out.println("Verification success: " + verifyResponse.isSuccess());
            System.out.println("Message: " + verifyResponse.getMessage());
        } catch (Exception e) {
            System.err.println("Gesture captcha error: " + e.getMessage());
        }
    }

    private static void jigsawCaptchaExample(CaptchaClient client) {
        System.out.println("\n=== Jigsaw Captcha Example ===");
        try {
            JigsawCaptchaResponse captcha = client.getJigsawCaptcha(300, 300, 3);
            System.out.println("Session ID: " + captcha.getSessionId());
            System.out.println("Image URL: " + captcha.getImageUrl());
            System.out.println("Grid size: " + captcha.getGridSize());
            System.out.println("Number of pieces: " + captcha.getPieces().size());

            List<JigsawPiece> pieces = captcha.getPieces();
            for (JigsawPiece piece : pieces) {
                piece.setCurrentX(piece.getOriginalX());
                piece.setCurrentY(piece.getOriginalY());
            }

            VerifyCaptchaResponse verifyResponse = client.verifyJigsawCaptcha(
                captcha.getSessionId(),
                pieces
            );

            System.out.println("Verification success: " + verifyResponse.isSuccess());
            System.out.println("Message: " + verifyResponse.getMessage());
        } catch (Exception e) {
            System.err.println("Jigsaw captcha error: " + e.getMessage());
        }
    }

    private static void voiceCaptchaExample(CaptchaClient client) {
        System.out.println("\n=== Voice Captcha Example ===");
        try {
            VoiceCaptchaResponse captcha = client.getVoiceCaptcha("zh-CN");
            System.out.println("Session ID: " + captcha.getSessionId());
            System.out.println("Audio URL: " + captcha.getAudioUrl());
            System.out.println("Text: " + captcha.getText());

            VerifyCaptchaResponse verifyResponse = client.verifyVoiceCaptcha(
                captcha.getSessionId(),
                captcha.getText()
            );

            System.out.println("Verification success: " + verifyResponse.isSuccess());
            System.out.println("Message: " + verifyResponse.getMessage());
        } catch (Exception e) {
            System.err.println("Voice captcha error: " + e.getMessage());
        }
    }

    private static void connectCaptchaExample(CaptchaClient client) {
        System.out.println("\n=== Connect Captcha Example ===");
        try {
            ConnectCaptchaResponse captcha = client.getConnectCaptcha();
            System.out.println("Session ID: " + captcha.getSessionId());
            System.out.println("Image URL: " + captcha.getImageUrl());

            List<List<Integer>> connections = Arrays.asList(
                Arrays.asList(0, 3),
                Arrays.asList(1, 4),
                Arrays.asList(2, 5)
            );

            VerifyCaptchaResponse verifyResponse = client.verifyConnectCaptcha(
                captcha.getSessionId(),
                connections
            );

            System.out.println("Verification success: " + verifyResponse.isSuccess());
            System.out.println("Message: " + verifyResponse.getMessage());
        } catch (Exception e) {
            System.err.println("Connect captcha error: " + e.getMessage());
        }
    }

    private static void threeDCaptchaExample(CaptchaClient client) {
        System.out.println("\n=== 3D Captcha Example ===");
        try {
            ThreeDCaptchaResponse captcha = client.getThreeDCaptcha();
            System.out.println("Session ID: " + captcha.getSessionId());
            System.out.println("Scene URL: " + captcha.getSceneUrl());
            System.out.println("Target ID: " + captcha.getTargetId());
            System.out.println("Hint: " + captcha.getHint());

            List<Double> targetPosition = Arrays.asList(100.5, 150.0, 50.0);
            VerifyCaptchaResponse verifyResponse = client.verifyThreeDCaptcha(
                captcha.getSessionId(),
                targetPosition
            );

            System.out.println("Verification success: " + verifyResponse.isSuccess());
            System.out.println("Message: " + verifyResponse.getMessage());
        } catch (Exception e) {
            System.err.println("3D captcha error: " + e.getMessage());
        }
    }

    private static void loginExample(CaptchaClient client) {
        System.out.println("\n=== Login Example ===");
        try {
            LoginResponse loginResponse = client.login("username", "password");
            System.out.println("Login successful");
            System.out.println("Access token: " + loginResponse.getAccessToken());
            System.out.println("Refresh token: " + loginResponse.getRefreshToken());
            System.out.println("Expires in: " + loginResponse.getExpiresIn() + " seconds");
            System.out.println("User: " + loginResponse.getUser().getUsername());

            client.logout();
            System.out.println("Logged out successfully");
        } catch (Exception e) {
            System.err.println("Login error: " + e.getMessage());
        }
    }

    private static void detectionExample(CaptchaClient client) {
        System.out.println("\n=== Detection Example ===");
        try {
            String script = client.getDetectionScript();
            System.out.println("Detection script length: " + script.length() + " characters");

            System.out.println("\n=== Done ===");
        } catch (Exception e) {
            System.err.println("Detection error: " + e.getMessage());
        }
    }
}

File: sdk/php/examples/UsageExample.php
<?php

declare(strict_types=1);

namespace Hjtpx\Captcha\Examples;

use Hjtpx\Captcha\Client\CaptchaClient;
use Hjtpx\Captcha\Exceptions\CaptchaApiException;
use Hjtpx\Captcha\Exceptions\CaptchaException;
use Hjtpx\Captcha\Exceptions\CaptchaNetworkException;
use Hjtpx\Captcha\Exceptions\CaptchaValidationException;

require_once __DIR__ . '/../vendor/autoload.php';

class UsageExample
{
    private CaptchaClient $client;

    public function __construct()
    {
        $this->client = new CaptchaClient(
            baseUrl: 'http://localhost:8080',
            apiKey: 'your-api-key',
            timeout: 30,
            maxRetries: 3
        );
    }

    public function sliderCaptchaExample(): void
    {
        echo "=== Slider Captcha Example ===\n";

        try {
            $captcha = $this->client->getSliderCaptcha(320, 160, 8);
            echo "Session ID: " . ($captcha['session_id'] ?? 'N/A') . "\n";
            echo "Image URL: " . ($captcha['image_url'] ?? 'N/A') . "\n";
            echo "Secret Y: " . ($captcha['secret_y'] ?? 'N/A') . "\n";

            $trajectory = [
                ['x' => 0, 'y' => $captcha['secret_y'] ?? 50, 't' => time() * 1000 - 1000],
                ['x' => 50, 'y' => ($captcha['secret_y'] ?? 50) + 5, 't' => time() * 1000 - 800],
                ['x' => 100, 'y' => ($captcha['secret_y'] ?? 50) - 3, 't' => time() * 1000 - 500],
                ['x' => 150, 'y' => ($captcha['secret_y'] ?? 50) + 2, 't' => time() * 1000 - 200],
                ['x' => 185, 'y' => $captcha['secret_y'] ?? 50, 't' => time() * 1000],
            ];

            $result = $this->client->verifySliderCaptcha(
                $captcha['session_id'],
                185,
                $captcha['secret_y'] ?? null,
                $trajectory
            );

            echo "Verification Success: " . ($result['success'] ? 'Yes' : 'No') . "\n";
            echo "Message: " . ($result['message'] ?? 'N/A') . "\n";
        } catch (CaptchaApiException $e) {
            echo "API Error: " . $e->getMessage() . " (Code: " . $e->getErrorCode() . ")\n";
        } catch (CaptchaNetworkException $e) {
            echo "Network Error: " . $e->getMessage() . "\n";
        } catch (CaptchaValidationException $e) {
            echo "Validation Error: " . $e->getMessage() . "\n";
        } catch (CaptchaException $e) {
            echo "Error: " . $e->getMessage() . "\n";
        }
    }

    public function clickCaptchaExample(): void
    {
        echo "\n=== Click Captcha Example ===\n";

        try {
            $captcha = $this->client->getClickCaptcha('number', 3, true);
            echo "Session ID: " . ($captcha['session_id'] ?? 'N/A') . "\n";
            echo "Image URL: " . ($captcha['image_url'] ?? 'N/A') . "\n";
            echo "Hint: " . ($captcha['hint'] ?? 'N/A') . "\n";
            echo "Hint Order: " . implode(', ', $captcha['hint_order'] ?? []) . "\n";

            $points = [[100, 100], [200, 150], [150, 200]];
            $clickSequence = [1, 2, 3];

            $result = $this->client->verifyClickCaptcha(
                $captcha['session_id'],
                $points,
                $clickSequence
            );

            echo "Verification Success: " . ($result['success'] ? 'Yes' : 'No') . "\n";
        } catch (\Throwable $e) {
            echo "Error: " . $e->getMessage() . "\n";
        }
    }

    public function imageCaptchaExample(): void
    {
        echo "\n=== Image Captcha Example ===\n";

        try {
            $captcha = $this->client->getImageCaptcha('mixed', 4);
            echo "Challenge ID: " . ($captcha['challenge_id'] ?? 'N/A') . "\n";
            echo "Image (base64): " . substr($captcha['image'] ?? '', 0, 50) . "...\n";

            $result = $this->client->verifyImageCaptcha(
                $captcha['challenge_id'],
                'ABCD'
            );

            echo "Verification Success: " . ($result['success'] ? 'Yes' : 'No') . "\n";
        } catch (\Throwable $e) {
            echo "Error: " . $e->getMessage() . "\n";
        }
    }

    public function batchVerifyExample(): void
    {
        echo "\n=== Batch Verify Example ===\n";

        try {
            $requests = [
                ['session_id' => 'session-1', 'x' => 100, 'y' => 50],
                ['session_id' => 'session-2', 'x' => 150, 'y' => 60],
                ['session_id' => 'session-3', 'x' => 200, 'y' => 70],
            ];

            $result = $this->client->batchVerify($requests);

            echo "Success Count: " . ($result['success_count'] ?? 0) . "\n";
            echo "Failed Count: " . ($result['failed_count'] ?? 0) . "\n";
            echo "Total Time: " . ($result['total_time_ms'] ?? 0) . "ms\n";

            foreach ($result['results'] ?? [] as $r) {
                echo "- Session " . ($r['session_id'] ?? 'N/A') . ": " .
                     ($r['success'] ? 'Success' : 'Failed') . "\n";
            }
        } catch (\Throwable $e) {
            echo "Error: " . $e->getMessage() . "\n";
        }
    }

    public function asyncVerifyExample(): void
    {
        echo "\n=== Async Verify Example ===\n";

        try {
            $asyncResult = $this->client->asyncVerify([
                'session_id' => 'session-async-1',
                'x' => 150,
                'y' => 50,
                'callback_url' => 'https://example.com/callback',
            ]);

            $taskId = $asyncResult['task_id'] ?? 'N/A';
            echo "Task ID: " . $taskId . "\n";
            echo "Status: " . ($asyncResult['status'] ?? 'N/A') . "\n";

            $pollCount = 0;
            while ($pollCount < 10) {
                $result = $this->client->getAsyncResult($taskId);
                $status = $result['status'] ?? '';

                echo "Poll $pollCount: Status = $status\n";

                if ($status === 'completed' || $status === 'failed') {
                    if ($status === 'completed' && isset($result['result'])) {
                        echo "Final Result: " . json_encode($result['result']) . "\n";
                    }
                    break;
                }

                usleep(500000);
                $pollCount++;
            }
        } catch (\Throwable $e) {
            echo "Error: " . $e->getMessage() . "\n";
        }
    }

    public function authenticationExample(): void
    {
        echo "\n=== Authentication Example ===\n";

        try {
            $result = $this->client->login('username', 'password');
            echo "Access Token: " . substr($result['access_token'] ?? '', 0, 20) . "...\n";
            echo "Expires In: " . ($result['expires_in'] ?? 0) . "s\n";
            echo "User: " . ($result['user']['username'] ?? 'N/A') . "\n";

            $this->client->logout();
            echo "Logged out successfully\n";
        } catch (\Throwable $e) {
            echo "Error: " . $e->getMessage() . "\n";
        }
    }

    public function detectionScriptExample(): void
    {
        echo "\n=== Detection Script Example ===\n";

        try {
            $script = $this->client->getDetectionScript('handleDetectionResult');
            echo "Script Length: " . strlen($script) . " bytes\n";
            echo "Script Preview: " . substr($script, 0, 100) . "...\n";
        } catch (\Throwable $e) {
            echo "Error: " . $e->getMessage() . "\n";
        }
    }
}

if (php_sapi_name() === 'cli' && basename(__FILE__) === basename($_SERVER['SCRIPT_NAME'] ?? '')) {
    $example = new UsageExample();

    echo "========================================\n";
    echo "   Hjtpx Captcha PHP SDK v15.0 Examples\n";
    echo "========================================\n\n";

    $example->sliderCaptchaExample();
    $example->clickCaptchaExample();
    $example->imageCaptchaExample();
    $example->batchVerifyExample();
    $example->asyncVerifyExample();
    $example->authenticationExample();
    $example->detectionScriptExample();

    echo "\n========================================\n";
    echo "             Examples Complete\n";
    echo "========================================\n";
}
